python/paddle/fluid/variable_index.py

```diff
@@ -139,19 +139,34 @@ def _getitem_impl_(var, item):
             end = MAX_INTEGER if end is None else end
 
         elif isinstance(slice_item, list):
+            is_bool_list = False
             for i in slice_item:
-                if not isinstance(i, int):
-                    raise TypeError("Only support int value in list")
+                if not isinstance(i, (int, bool)):
+                    raise TypeError("Only support int or bool in list.")
```

Comment:
"Only support int or bool in list" -> "Only support int or bool in index list"

Otherwise users may be confused about the list thing, them may not know we refer to the index list.

Reply:
Thanks, done.